### PR TITLE
Add more check for array_difference function

### PR DESCRIPTION
**Pull request by kangkaisen**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

For https://github.com/StarRocks/starrocks/issues/4336

1. Add more check for array_difference function
2. Change equalsIgnoreCase to equals in FunctionAnalyzer

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 44 / 44 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ArrayType.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/FunctionAnalyzer.java|43|43|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java [hunk 1]
```diff
@@ -108,12 +119,12 @@ private static void analyzeBuiltinAggFunction(FunctionCallExpr functionCallExpr)
             }
         }
 
-        if (fnName.getFunction().equalsIgnoreCase("dense_rank")
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:111]: has `fnName` already convert to low case?

**Comment by HangyuanLiu** [comment]
Why can we use equal instead of equalsIgnoreCase, and where do we currently ensure lowercase function names? Where is the code entry?

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java [hunk 2]
```diff
@@ -218,26 +229,23 @@ private static void analyzeBuiltinAggFunction(FunctionCallExpr functionCallExpr)
...
         }
 
-        if (fnName.getFunction().equalsIgnoreCase("min")
-                || fnName.getFunction().equalsIgnoreCase("max")) {
+        if (fnName.getFunction().equals(FunctionSet.MIN)
+                || fnName.getFunction().equals(FunctionSet.MAX)
+                || fnName.getFunction().equals(FunctionSet.NDV)
+                || fnName.getFunction().equals(FunctionSet.HLL_UNION_AGG)) {
             fnParams.setIsDistinct(false);  // DISTINCT is meaningless here
-        } else if (fnName.getFunction().equalsIgnoreCase("DISTINCT_PC")
```
Comment (trueeyu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:233]: what is `DISTINCT_PC` ?

**Review by liuyehcf — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java (hunk 1 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:111]: Yes. All function names should be lower case.

**Comment by kangkaisen** [comment]
> Why can we use equal instead of equalsIgnoreCase, and where do we currently ensure lowercase function names? Where is the code entry?

All function names should be lower case.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java (hunk 2 above)
Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java:233]: I don't know,  but this function isn’t implemented always，so I remove  it.

**Review by HangyuanLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**